pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -159,8 +173,35 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 		orchestrationStatusKey, provisioningLabelKey,
 	)
 
-	originalNodes, remainingNodes, completed, statuses, err := c.checkNodesCompleted(selector, config, configOSDs)
+	// For OSDs that are definitely update operations, we want to be able to defer the update until
+	// after we have created new OSDs in order to speed up cluster scale-out. Allow us to defer
+	// updating by keeping a list of deferred configmaps and processing those after creates have
+	// been done. If OSD provisioning doesn't report "alreadyExists" status, then no configmaps will
+	// be deferred, and OSD create/update ops will be intermixed in whatever order is processed.
+	deferredConfigMaps := []*v1.ConfigMap{}
+	// handleDeferredConfigMaps NEEDS to happen before the outer function returns.
+	// The nested for loops make it hard to break out and handle the return value cleanly,
+	// so remember to call this before all return statements.
+	handleDeferredConfigMaps := func() {
```

Comment:
Why not define a separate function? I'm not a huge fan of nested functions unless maybe it's a one-liner or the outer context is really necessary..

Reply:
The nested loops were already defined as part of this function. I really don't like them either, but I'm nervous about breaking functionality by changing things.

Reply:
I meant defining a separate function, rather than the inline function. Or how does the loop affect the inline function?

Reply:
I see. I should be able to do this in my next update Monday morning.

Reply:
I think the outer context, especially the comment above, is pretty important for this function. I moved it locally, and I think some context is lost after doing so. I'm hoping to finish my design for https://github.com/rook/rook/pull/6975 and then start on those changes, which will mean a rewrite of this code anyway. If I thought this code would have to stay, I would really want to work to avoid the triple-nested-for loop situation going on here.

Reply:
That's fine to keep it, especially if the future PR might change it anyway.